### PR TITLE
Add tsb subscribe command

### PR DESCRIPTION
**Pull request by dsouza93**
Adds feature requested in #3. Users can now call `tsb subscribe {branch}`. This will ensure that the branch exists, then update the patches file to include the subscription.

Running `tsb subscribe {branch}` adds the branch tag to the specified repo in patches, and running `tsb update` will fetch all changesets between the configured repository head and the end of the branch, in order.

`tsb ls-cherry` has been updated to report subscription changesets as well.
`tsb build` has been updated to iterate through all subscription changesets in order after applying all user selected changesets. 

The README has been updated to reflect the changes.

### CODE REVIEW

File: cmd.go [hunk 1]
```diff
@@ -83,10 +83,23 @@ func (e *Executor) Execute() error {
...
-				if err != nil {
-					return fmt.Errorf("Unable to apply %s to %s: "+cherry, name, err.Error())
+			if cfg.Patches[name] != nil {
+				for _, cherry := range cfg.Patches[name].Changesets {
+					err = repo.Cherry(e.Dir(), name, cherry)
+					if err != nil {
+						return fmt.Errorf("Unable to apply %s to %s: "+cherry, name, err.Error())
+					}
+				}
+				subscriptions := cfg.Patches[name].Subscriptions
```
Comment (alficles) [inline cmd.go]: So, this means that all subscriptions are applied after all patches. This isn't necessarily always right. Sometimes you'll need individual patches before and after subscriptions.

Is it possible to roll it all into a single object like:

```    superwidget:
      changesets:
      - {changeset1}
      - branch: remote01/widget_feature
        changesets:
          - {changesetA}
          - {changesetB}
      - {changeset2}
```
?

File: README.md [hunk 2]
```diff
@@ -85,19 +90,32 @@ successful.
...
 
-The hashes can refer to any changeset in any of the referenced sources.
-Each changeset will be cherry-picked, in order, to the head of the
-branch to be built.
+    superwidget:
+      changesets:
+      - {changeset1}
+      - {changeset2}
+      subscriptions:
+          - branch: remote01/widget_feature
```
Comment (alficles) [inline README.md]: To what does "remote01" refer here?

We don't currently have a clear way to refer to extra branches; at the moment, everything assumes you're tracking the origin. I think we might need a way to talk about extra repos by name.

**Review by alficles — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (dsouza93) [reply README.md]: I've made a change to allow users to give names to the various extras in the repo config.

File: cmd.go (hunk 1 above)
Comment (dsouza93) [reply cmd.go]: Updated to be backward compatible and subscriptions/changesets are now interchangeable.

**Review by alficles — APPROVED** [review]

**Pull request merged**